include/net/fota_download.h

```diff
@@ -30,19 +30,31 @@ extern "C" {
  * @brief FOTA download event IDs.
  */
 enum fota_download_evt_id {
+	/** FOTA download progress report. */
+	FOTA_DOWNLOAD_EVT_PROGRESS,
 	/** FOTA download finished. */
 	FOTA_DOWNLOAD_EVT_FINISHED,
 	/** FOTA download error. */
 	FOTA_DOWNLOAD_EVT_ERROR,
 };
 
+/**
+ * @brief FOTA download event data.
+ */
+struct fota_download_evt {
+	enum fota_download_evt_id id;
+#ifdef CONFIG_FOTA_DOWNLOAD_PROGRESS_EVT
```

Comment:
Why this ifdef?

Reply:
so that the structure is smaller if you disable the progress evt. as it would be an unused field.